tracer/src/Datadog.Trace/Agent/StatsAggregator.cs

```diff
@@ -118,7 +128,27 @@ internal async Task Flush()
             // Use a do/while loop to still flush once if _processExit is already completed (this makes testing easier)
             do
             {
-                await Task.WhenAny(_processExit.Task, Task.Delay(_bucketDuration)).ConfigureAwait(false);
+                var delayTask = Task.Delay(_bucketDuration);
+                var sendStats = true;
+
+                // Until the initialization task has finished, check it on each iteration
+                if (CanComputeStats is null)
+                {
+                    var completedTask = await Task.WhenAny(_initializationTask, _processExit.Task, delayTask).ConfigureAwait(false);
+
+                    if (completedTask != _initializationTask)
+                    {
+                        // The initialization task hasn't completed yet so we don't know if we can compute stats
+                        sendStats = false;
```

Comment:
if we don't know, should we return here? I assume the stats will be sent by the agent when we send the corresponding traces, no?

Reply:
I think I avoided this in the new PR, but let me know if you have similar concerns in #3152